### PR TITLE
fix: Dashboard chart UI issues

### PR DESCRIPTION
**Pull request by omkarK06**
#4256 

## Summary by CodeRabbit

- **New Features**
	- Introduced new functions for enhanced chart rendering: `getLargestLabel` for improved label sizing, `largestStackLabel` for stacked charts, and `getPieChartRadius` for pie charts.

- **Improvements**
	- Enhanced axis label handling for better visual representation in horizontal bar and stacked charts.
	- Updated text width calculation to be more flexible with an optional font size parameter.
	- Improved clarity in variable naming for better understanding.
	- Enhanced readability of the ViewPanel component template and improved management of the loading state.
	- Refined UI test interactions with more descriptive data-test selectors, enhancing maintainability.
	- Improved handling of large datasets and complex SQL queries in tests.

- **Bug Fixes**
	- Enhanced error handling in tests for invalid queries, improving overall robustness of the application.

### CODE REVIEW

File: web/src/utils/dashboard/convertSQLData.ts [hunk 1]
```diff
@@ -430,6 +430,60 @@ export const convertSQLData = async (
...
+    const radius = minRadius / 2;
+
+    let multiplier = 110;
+
+    if (radius > 90) multiplier = 130;
+
+    if (radius > 150) multiplier = 150;
+
+    return (radius / minRadius) * multiplier;
+  };
```
[the remark this answers is not in the document]
Comment (gaby) [reply web/src/utils/dashboard/convertSQLData.ts:485]: @omkarK06 This is probably needed

File: web/src/utils/dashboard/convertSQLData.ts [hunk 2]
```diff
@@ -707,7 +761,10 @@ export const convertSQLData = async (
             ? largestLabel(getAxisDataFromKey(yAxisKeys[0]))
             : formatUnitValue(
                 getUnitValue(
-                  largestLabel(getAxisDataFromKey(yAxisKeys[0])),
+                  panelSchema.type === "stacked" ||
+                    panelSchema.type === "area-stacked"
+                    ? largestStackLabel(yAxisKeys[0], breakDownKeys[0])
+                    : largestLabel(getAxisDataFromKey(yAxisKeys[0])),
```
[the remark this answers is not in the document]
Comment (gaby) [reply web/src/utils/dashboard/convertSQLData.ts:767]: @omkarK06 This too

File: web/src/utils/dashboard/convertSQLData.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/utils/dashboard/convertSQLData.ts]: Fixed

File: web/src/utils/dashboard/convertSQLData.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/utils/dashboard/convertSQLData.ts]: Fixed

**Review by ktx-abhay — APPROVED** [review]

**Pull request merged**